### DISCUSSION TITLE
Automerge multiple Pull Requests on same repository

### DISCUSSION THREAD

**Discussion by trietsch** [renovatebot/renovate#14061] [Request Help]
I've set up Renovate a configuration that basically can create the following PRs on a repository:
1. NPM dev dependencies, `autoMerge` enabled
2. Gradle Wrapper Major+Minor+Patch, `autoMerge` enabled
3. Gradle App Dependencies Minor+Patch, `autoMerge` enabled
4. Gradle App Dependencies Major, `autoMerge` **disabled**
5. Gradle Build Dependencies Major+Minor+Patch, `autoMerge` enabled
6. Terraform Minor+Patch, `autoMerge` enabled
7. Terraform Major, `autoMerge` **disabled**

This means, that for a single Renovate run over this repository, 7 PRs can be created potentially. CI is setup for this repository and includes tests.

Basically, when Renovate scans this repository and determines which PRs should be created, 7 CI pipelines are started simultaneously. The pipeline that finishes **first** and is **green**, will be merged automatically. But the other 6 pipelines are now stale, and need to wait on the next Renovate run to rebase them (rebase is done when a branch is behind the base branch).

Any tips on how to handle this _race of PRs_? I'd like to merge all branches automatically that are green. And actually, NPM dev dependencies can even be committed to the base branch, if that's possible.

**Comment by rarkins** [comment]
Use `prConcurrentLimit` to reduce open PR count if you're doing a lot of automerging, that will reduce the amount of rebasing and wasted tests. Renovate won't automerge unless branches are up to date.

**Reply by trietsch** [reply]
Will try it out, thanks!
